homeassistant/components/vasttrafik/sensor.py

```diff
@@ -139,11 +144,16 @@ def update(self):
             self._state = None
             self._attributes = {}
         else:
+            skip_index = 0
             for departure in self._departureboard:
                 line = departure.get("sname")
                 if "cancelled" in departure:
                     continue
                 if not self._lines or line in self._lines:
+                    if skip_index < self._skip:
```

Comment:
If the one is cancelled, you don't increment the skip_index. Is it as expected ?
Perhaps you should count the cancelled in the skip_index.

Reply:
That's expected. Let say I have the config in my example;
```
          departures:
                  - from: "Båtsman Kapers gata"
                    heading: "Körkarlens gata"
                    name: "Next bus"
                  - from: "Båtsman Kapers gata"
                    heading: "Körkarlens gata"
                    name: "Next bus thereafter"
                    skip: 1
```
If  the next scheduled bus is cancelled the "Next bus" entity gets the next scheduled bus. 
If I don't take canceled departures into account then "Next bus" and "Next bus thereafter" will show the same departure.
Does that make sense to you?

Reply:
OK I think I understand now.